lms/djangoapps/edraak_marketing_email/helpers.py

```diff
@@ -1,9 +1,25 @@
+from django.conf import settings
 from edraak_marketing_email.models import UnsubscribedUser
+from edraak_sendinblue.api_client import create_contact
 
 
 def unsubscribe_from_marketing_emails(user):
     UnsubscribedUser.objects.get_or_create(user=user)
+    change_sendinblue_user_state(user)
 
 
 def subscribe_to_marketing_emails(user):
     UnsubscribedUser.objects.filter(user=user).delete()
+    change_sendinblue_user_state(user)
+
+
+def change_sendinblue_user_state(user):
+    if hasattr(settings, "EDRAAK_SENDINBLUE_API_KEY") and settings.EDRAAK_SENDINBLUE_API_KEY:
+        blacklisted = not UnsubscribedUser.is_user_subscribed(user=user)
+
+        create_contact(
+            name=user.profile.name,
+            email=user.email,
+            blacklisted=blacklisted,
+            username=user.username
+        )
```

Comment:
Here you are creating a contact, what you need is updating a contact. So you need to create an update function here` lms/djangoapps/edraak_sendinblue/api_client.py` and use that here.

Best is to create an update_or_create function, whereby an update request is sent to sendinblue api, if the user doesnt exist then create the contact with the proper subscription status whether subscribed or non subscribed

check these:
https://developers.sendinblue.com/docs/available-functions-in-api-clients
https://github.com/sendinblue/APIv3-python-library/
https://developers.sendinblue.com/docs/how-it-works
*in php*: https://github.com/sendinblue/APIv3-php-library/blob/master/docs/Api/ContactsApi.md#updatecontact